managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -35,6 +35,7 @@
 import io.netty.util.Recycler;
 import io.netty.util.Recycler.Handle;
 import java.time.Clock;
+import java.util.ArrayList;
```

Comment:
Is it required?

Reply:
If it fixed a build issue, I think we'd better have a separate PR.

Reply:
The branch keeps cherry-picking new PRs without building success. I have pushed one PR to fix the build, https://github.com/apache/pulsar/pull/19645, and the CI needs to wait for a long time and rerun a few times to pass.

Let me clean up the build-related changes to see whether we can build success.

Reply:
The branch-2.9 still build failed.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project managed-ledger: Compilation failure
[ERROR] /Users/hangc/Workspace/sourcecode/CR/pulsar/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:[436,85] cannot find symbol
[ERROR]   symbol:   class ArrayList
[ERROR]   location: class org.apache.bookkeeper.mledger.impl.ManagedLedgerImpl

```

Reply:
Pushed one PR to fix it. https://github.com/apache/pulsar/pull/19669. After that PR is merged, I will rebase branch-2.9